django/http/request.py

```diff
@@ -230,6 +288,17 @@ def body(self):
         if not hasattr(self, '_body'):
             if self._read_started:
                 raise RawPostDataException("You cannot access body after reading from request's data stream")
+
```

Comment:
>Why would you not want to protect against a naked read() that will gobble the whole request into memory, no matter the content-type?

Because you would limit the allowed upload file size, which is not read directly in memory. The whole idea of both new parameters is to not load huge stuff in a dict, or similar structure where it consumes much more memory and CPU.

Note that the MultiPartPerser has its own logic so as not to read everything in memory. Am I wrong?